tests/app/App.test.tsx

```diff
@@ -0,0 +1,21 @@
+import { describe } from "vitest";
+import "@testing-library/jest-dom";
+import { render } from "../test-utlis";
+import App from "../../src/App";
+import { fireEvent, screen } from "@testing-library/react";
+
+describe("react router functionality", async () => {
+  it("should navigate between two routes and render corresponding components", () => {
+    render(<App />);
+
+    expect(screen.getByText("hej, tu PlantMomma")).toBeInTheDocument();
+
+    const linkElement = screen.getByText("Zarejestruj");
+
+    fireEvent.click(linkElement);
+
+    screen.debug();
+
+    expect(screen.getByText("nazwa użytkownika"));
+  });
```

Comment:
you gave a proper description of the test case so I understand what you are doing. But in a slightly longer test, I'd be lost.

You can put all selections into variables and name them in such a way so that it's obvious that one element is displayed on the first location and the other only appears on the register page.